src/main/java/com/synopsys/integration/alert/provider/blackduck/web/PolicyNotificationFilterCustomEndpoint.java

```diff
@@ -0,0 +1,113 @@
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.common.action.CustomEndpointManager;
+import com.synopsys.integration.alert.common.descriptor.config.field.endpoint.table.TableSelectCustomEndpoint;
+import com.synopsys.integration.alert.common.descriptor.config.ui.ProviderDistributionUIConfig;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.rest.ResponseFactory;
+import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
+import com.synopsys.integration.alert.provider.blackduck.BlackDuckProperties;
+import com.synopsys.integration.alert.provider.blackduck.descriptor.BlackDuckDescriptor;
+import com.synopsys.integration.blackduck.api.generated.enumeration.NotificationType;
+import com.synopsys.integration.blackduck.rest.BlackDuckHttpClient;
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.blackduck.service.PolicyRuleService;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+@Component
+public class PolicyNotificationFilterCustomEndpoint extends TableSelectCustomEndpoint {
+    private final Logger logger = LoggerFactory.getLogger(PolicyNotificationFilterCustomEndpoint.class);
+    private BlackDuckProperties blackDuckProperties;
+
+    @Autowired
+    protected PolicyNotificationFilterCustomEndpoint(CustomEndpointManager customEndpointManager, ResponseFactory responseFactory,
+        Gson gson, BlackDuckProperties blackDuckProperties) throws AlertException {
+        super(BlackDuckDescriptor.KEY_BLACKDUCK_POLICY_NOTIFICATION_TYPE_FILTER, customEndpointManager, responseFactory, gson);
+        this.blackDuckProperties = blackDuckProperties;
+    }
+
+    @Override
+    protected List<?> createData(Map<String, FieldValueModel> fieldValueModels) throws AlertException {
+        FieldValueModel fieldValueModel = fieldValueModels.get(ProviderDistributionUIConfig.KEY_NOTIFICATION_TYPES);
+
+        if (null == fieldValueModel) {
+            return List.of();
+        }
+
+        Collection<String> selectedNotificationTypes = fieldValueModel.getValues();
+        if (selectedNotificationTypes.isEmpty()) {
+            return List.of();
+        }
+
+        if (isFilterablePolicy(selectedNotificationTypes)) {
+            try {
+                return retrieveBlackDuckPolicyOptions();
+            } catch (IntegrationException e) {
+                logger.error("There was an issue communicating with Black Duck");
+                logger.debug(e.getMessage(), e);
+                throw new AlertException("Was unable to communicate with Black Duck.", e);
```

Comment:
To keep it grammatically correct, drop the "Was": "Unable to communicate with Black Duck."

Reply:
Fixed it in my branch that adds to this.